### PR TITLE
fix shellcheck pass in sample-apiserver

### PR DESCRIPTION
**Pull request by danielqsj**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

fix shellcheck failures in `staging/src/k8s.io/sample-apiserver`

**Which issue(s) this PR fixes**:

work towards #72956

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Review by BenTheElder — APPROVED** [review]
/lgtm
thanks!

**Comment by danielqsj** [comment]
@deads2k if you have time, can you help review this? Thanks

**Comment by danielqsj** [comment]
@sttts fixed, PTAL

File: staging/src/k8s.io/sample-apiserver/hack/build-image.sh [hunk 1]
```diff
@@ -24,7 +24,7 @@ function cleanup {
 }
 trap cleanup EXIT
 
-pushd "${KUBE_ROOT}/vendor/k8s.io/sample-apiserver"
+pushd "${KUBE_ROOT}/vendor/k8s.io/sample-apiserver" || exit
```
[the remark this answers is not in the document]
Comment (danielqsj) [reply staging/src/k8s.io/sample-apiserver/hack/build-image.sh]: thanks, fixed. :)

[the remark this answers is not in the document]
Comment (sttts) [reply staging/src/k8s.io/sample-apiserver/hack/build-image.sh]: Actually, I agree with @fejta's comment in one of the other PRs: add `set -e` so this (and any other error) fails the script.

[the remark this answers is not in the document]
Comment (danielqsj) [reply staging/src/k8s.io/sample-apiserver/hack/build-image.sh]: can we add them both? `exit 1` and `set -e`

[the remark this answers is not in the document]
Comment (sttts) [reply staging/src/k8s.io/sample-apiserver/hack/build-image.sh]: Why should we?

[the remark this answers is not in the document]
Comment (danielqsj) [reply staging/src/k8s.io/sample-apiserver/hack/build-image.sh]: sure. thanks. I'm not familiar with this.

[the remark this answers is not in the document]
Comment (BenTheElder) [reply staging/src/k8s.io/sample-apiserver/hack/build-image.sh]: `set -e` is _roughly_ like having `|| exit 1` after everything, so doing both is pointless, we do `set -e` in most of our scripts.
see: https://www.gnu.org/software/bash/manual/html_node/The-Set-Builtin.html

File: staging/src/k8s.io/sample-apiserver/hack/update-codegen.sh [hunk 2]
```diff
@@ -18,24 +18,24 @@ set -o errexit
...
-CODEGEN_PKG=${CODEGEN_PKG:-$(cd ${SCRIPT_ROOT}; ls -d -1 ./vendor/k8s.io/code-generator 2>/dev/null || echo ../code-generator)}
+SCRIPT_ROOT=$(dirname "${BASH_SOURCE[0]}")/..
+CODEGEN_PKG=${CODEGEN_PKG:-$(cd "${SCRIPT_ROOT}"; ls -d -1 ./vendor/k8s.io/code-generator 2>/dev/null || echo ../code-generator)}
 
 # generate the code with:
 # --output-base    because this script should also be able to run inside the vendor dir of
 #                  k8s.io/kubernetes. The output-base is needed for the generators to output into the vendor dir
 #                  instead of the $GOPATH directly. For normal projects this can be dropped.
-${CODEGEN_PKG}/generate-groups.sh all \
+"${CODEGEN_PKG}"/generate-groups.sh all \
```
Comment (BenTheElder) [inline staging/src/k8s.io/sample-apiserver/hack/update-codegen.sh]: please quote the whole string 
```suggestion
"${CODEGEN_PKG}/generate-groups.sh" all \
```

File: staging/src/k8s.io/sample-apiserver/hack/update-codegen.sh [hunk 3]
```diff
@@ -18,24 +18,24 @@ set -o errexit
...
   k8s.io/sample-apiserver/pkg/client k8s.io/sample-apiserver/pkg/apis \
   "wardle:v1alpha1,v1beta1" \
-  --output-base "$(dirname ${BASH_SOURCE})/../../.." \
-  --go-header-file ${SCRIPT_ROOT}/hack/boilerplate.go.txt
+  --output-base "$(dirname "${BASH_SOURCE[0]}")/../../.." \
+  --go-header-file "${SCRIPT_ROOT}"/hack/boilerplate.go.txt
 
-${CODEGEN_PKG}/generate-internal-groups.sh "deepcopy,defaulter,conversion" \
+"${CODEGEN_PKG}"/generate-internal-groups.sh "deepcopy,defaulter,conversion" \
```
Comment (BenTheElder) [inline staging/src/k8s.io/sample-apiserver/hack/update-codegen.sh]: here too:
```suggestion
"${CODEGEN_PKG}/generate-internal-groups.sh" "deepcopy,defaulter,conversion" \
```

File: staging/src/k8s.io/sample-apiserver/hack/update-codegen.sh [hunk 4]
```diff
@@ -18,24 +18,24 @@ set -o errexit
...
   k8s.io/sample-apiserver/pkg/client k8s.io/sample-apiserver/pkg/apis k8s.io/sample-apiserver/pkg/apis \
   "wardle:v1alpha1,v1beta1" \
-  --output-base "$(dirname ${BASH_SOURCE})/../../.." \
-  --go-header-file ${SCRIPT_ROOT}/hack/boilerplate.go.txt
+  --output-base "$(dirname "${BASH_SOURCE[0]}")/../../.." \
+  --go-header-file "${SCRIPT_ROOT}"/hack/boilerplate.go.txt
```
Comment (BenTheElder) [inline staging/src/k8s.io/sample-apiserver/hack/update-codegen.sh]: and here:
```suggestion
  --go-header-file "${SCRIPT_ROOT}/hack/boilerplate.go.txt"
```

**Comment by danielqsj** [comment]
@BenTheElder fixed, thanks :)

**Review by BenTheElder — APPROVED** [review]
/lgtm

**Pull request merged**